fsharp-backend/src/LibExecution/Legacy.fs

```diff
@@ -0,0 +1,155 @@
+            Map.iter
+              (fun k v ->
+                w.WritePropertyName k
+                writeDval v)
+              o)
+    | DFnVal _ ->
+        // See docs/dblock-serialization.md
+        wrapNullValue "block"
+    | DIncomplete _ -> wrapNullValue "incomplete"
+    | DChar c -> wrapStringValue "character" c
+    | DError (_, msg) -> wrapStringValue "error" msg
+    | DHttpResponse (h) ->
+        w.writeObject
+          (fun () ->
+            w.WritePropertyName "type"
+            w.WriteValue "response"
+            w.WritePropertyName "value"
+
+            w.writeArray
+              (fun () ->
+                match h with
+                | Redirect str ->
+                    w.writeArray
+                      (fun () ->
+                        w.WriteValue "Redirect"
+                        w.WriteValue str)
+
+                    writeDval DNull
+                | Response (code, headers, hdv) ->
+                    w.writeArray
+                      (fun () ->
+                        w.WriteValue "Response"
+                        w.WriteValue code
+
+                        w.writeArray
+                          (fun () ->
+                            List.iter
+                              (fun (k : string, v : string) ->
+                                w.writeArray
+                                  (fun () ->
+                                    w.WriteValue k
+                                    w.WriteValue v))
+                              headers))
+
+                    writeDval hdv))
+    | DDB dbname -> wrapStringValue "datastore" dbname
+    | DDate date -> wrapStringValue "date" (date.toIsoString ())
+    | DPassword (Password hashed) ->
+        if redact then
+          wrapNullValue "password"
+        else
+          hashed |> base64Encode |> wrapStringValue "password"
+    | DUuid uuid -> wrapStringValue "uuid" (uuid.ToString())
+    | DOption opt ->
+        (match opt with
+         | None -> wrapNullValue "option"
+         | Some ndv -> wrapNestedDvalValue "option" ndv)
+    | DErrorRail erdv -> wrapNestedDvalValue "errorrail" erdv
+    | DResult res ->
+        (match res with
+         | Ok rdv ->
+             w.writeObject
+               (fun () ->
+                 w.WritePropertyName "type"
+                 w.WriteValue("result")
+                 w.WritePropertyName "constructor"
+                 w.WriteValue("Ok")
+                 w.WritePropertyName "values"
+                 w.writeArray (fun () -> writeDval rdv))
+         | Error rdv ->
+             w.writeObject
+               (fun () ->
+                 w.WritePropertyName "type"
+                 w.WriteValue("result")
+                 w.WritePropertyName "constructor"
+                 w.WriteValue("Error")
+                 w.WritePropertyName "values"
+                 w.writeArray (fun () -> writeDval rdv)))
+    | DBytes bytes -> wrapStringValue "bytes" (System.Convert.ToBase64String bytes)
```

Comment:
In legacy, this was not base64 encoded, AFAICT. So I think this should be:

`bytes |> System.Text.Encoding.ASCII.GetString |> wrapStringValue "bytes"`